assets/js/script.js

```diff
@@ -0,0 +1,101 @@
+const carouselList = document.querySelector(".carousel-list"),
+  carouselItems = document.querySelectorAll(".carousel-list__item"),
+  carouselWrap = document.querySelector(".carousel-wrap"),
+  btnNext = document.querySelector(".carousel-next"),
+  btnPrev = document.querySelector(".carousel-prev"),
+  pagination = document.querySelectorAll(".pagination-bullet");
+
+let carouselWrapWidth = carouselWrap.offsetWidth,
+  carouselItemsLength = carouselItems.length,
+  carouselIemsActiveIndex = 0,
+  isResizing = false;
+
+// Add or disable switch transition
+function addTransition() {
+  !isResizing
+    ? (carouselList.style.transition = "transform 0.5s ease-in-out")
+    : (carouselList.style.transition = "");
+}
+
+// Set active class to carousel item
+function addActiveToItem() {
+  carouselItems.forEach((item, index) => {
+    index === carouselIemsActiveIndex
+      ? item.classList.add("carousel-list__item--active")
+      : item.classList.remove("carousel-list__item--active");
+  });
+}
+// Add/remove class to pagination bullet
+function setActivePagination() {
+  pagination.forEach((item, index) => {
+    index === carouselIemsActiveIndex
+      ? item.classList.add("pagination-bullet--active")
+      : item.classList.remove("pagination-bullet--active");
+  });
+}
+
+// Navigate carousel on pagination click
+function moveCarouselOnPagination() {
+  pagination.forEach((item, index) => {
+    item.addEventListener("click", () => {
```

Comment:
Just a suggestion to maybe look into using event delegation: registering a click listener on the `pagination` wrapper and in the handler checking, which child was actually clicked.  
This would allow us to potentially add new carousel items without the need of additional listeners.

Reply:
Yeah, had a thought to use event delegation, but choose more simple way to do it. Extra mile doent hurt anyone yet. Thanks, will remember that.